diff --git a/dist/utils/keys.js b/dist/utils/keys.js
index 289330c073270493c040e8d043ae0ce1e8916cc9..cc3fd2d852dec3a28f3fd3b277f88323682c717a 100644
--- a/dist/utils/keys.js
+++ b/dist/utils/keys.js
@@ -7,7 +7,7 @@ function inferKeyType(owner) {
     if (typeof owner === 'string') {
         owner = (0, serial_1.hexToBytes)(owner);
     }
-    if (owner.length === 32) {
+    if (owner.length === 32 || owner.length === 33) {
         return enums_1.AccountKeyType.ED25519;
     }
     if (owner.length === 20) {
