diff --git a/dist/production/index.js b/dist/production/index.js
index 7494b25c43c08069b9d990dd1692bd5262314e3b..ffd44be748835424c803356d402736d5fb5995ee 100644
--- a/dist/production/index.js
+++ b/dist/production/index.js
@@ -10958,7 +10958,7 @@ var createRequestHandler = (build, mode) => {
       });
     };
     if (_build.future.unstable_middleware) {
-      if (initialContext == null) {
+      if (!(initialContext instanceof Map)) {
         loadContext = new unstable_RouterContextProvider();
       } else {
         try {
