diff --git a/dist/development/index.js b/dist/development/index.js
index dce641e586e41f64e1be8d45fb54eadf6a468ff4..ea5ce3521360504052b8cf60c15698aba8590e58 100644
--- a/dist/development/index.js
+++ b/dist/development/index.js
@@ -1210,14 +1210,11 @@ function derive(build, mode) {
       });
     };
     if (build.future.v8_middleware) {
-      if (initialContext && !(initialContext instanceof _chunkWICQJKU6js.RouterContextProvider)) {
-        let error = new Error(
-          "Invalid `context` value provided to `handleRequest`. When middleware is enabled you must return an instance of `RouterContextProvider` from your `getLoadContext` function."
-        );
-        handleError(error);
-        return returnLastResortErrorResponse(error, serverMode);
+      const map = new Map();
+      for (const key in initialContext) {
+         map.set(key, initialContext[key]);
       }
-      loadContext = initialContext || new (0, _chunkWICQJKU6js.RouterContextProvider)();
+      loadContext = initialContext || new (0, _chunkWICQJKU6js.RouterContextProvider)(map);
     } else {
       loadContext = initialContext || {};
     }
diff --git a/dist/production/index.js b/dist/production/index.js
index 439b88cdabc6492147302ac0015c682aa0c1953f..7eedda74b605124e5e51e7c5b7c2879462a85f75 100644
--- a/dist/production/index.js
+++ b/dist/production/index.js
@@ -1210,14 +1210,11 @@ function derive(build, mode) {
       });
     };
     if (build.future.v8_middleware) {
-      if (initialContext && !(initialContext instanceof _chunkQSNCZFX5js.RouterContextProvider)) {
-        let error = new Error(
-          "Invalid `context` value provided to `handleRequest`. When middleware is enabled you must return an instance of `RouterContextProvider` from your `getLoadContext` function."
-        );
-        handleError(error);
-        return returnLastResortErrorResponse(error, serverMode);
+      const map = new Map();
+      for (const key in initialContext) {
+        map.set(key, initialContext[key]);
       }
-      loadContext = initialContext || new (0, _chunkQSNCZFX5js.RouterContextProvider)();
+      loadContext = initialContext || new (0, _chunkQSNCZFX5js.RouterContextProvider)(map);
     } else {
       loadContext = initialContext || {};
     }
